lib/near-contract.js

```diff
@@ -2,19 +2,19 @@ import * as near from "./api";
 export class NearContract {
     deserialize() {
         const rawState = near.storageRead("STATE");
+        let c = this.default();
         if (rawState) {
             const state = JSON.parse(rawState);
-            // reconstruction of the contract class object from plain object
-            let c = this.default();
             Object.assign(this, state);
+            // reconstruction of the contract class object from plain object
             for (const item in c) {
                 if (c[item].constructor?.deserialize !== undefined) {
                     this[item] = c[item].constructor.deserialize(this[item]);
                 }
             }
         }
         else {
-            throw new Error("Contract state is empty");
+            Object.assign(this, c);
```

Comment:
maybe we could make this optional. If `default` is defined, then grab it and use it, if not throw error. Since not every contract might want to have a default.

Reply:
I believe there is such an option in RS SDK. This design does not allow developers to have an uninitialized contract.

Reply:
I see, you guys are right. In RS SDK, developer can right panic in `impl Default` to have a "must explicit initialize contract", but here it's not possible. If we throw error in `default()`, then because deserialization also calls `default()`, it will not work

Reply:
Looks like Serhii's `@Nearbindgen defaultState` proposal is a great solution to the problem, developer can configure the contract to allow or prohibit default there. We're shipping freeze for next week, so let's take more time to get consensus on the solution!